src/Tests/UnitTest/TestComponentCSharp_Tests.cs

```diff
@@ -528,7 +528,7 @@ public void TestBuffer()
 
         async Task TestStorageFileAsync()
         {
-            var folderPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            var folderPath = AppContext.BaseDirectory;
             StorageFile file = await StorageFile.GetFileFromPathAsync(folderPath + "\\UnitTest.dll");
```

Comment:
This is the problematic line the AppContext.BaseDirectory already contains a forward slash at the end of the path, for example `cswinrt\src\Tests\unittest\bin\x64\Release\net6.0\` 
Changing the code to
```C#
StorageFile file = await StorageFile.GetFileFromPathAsync($"{folderPath}UnitTest.dll");
```
worked for me. 
Having said that I think we might need something to test even this same test storage test in single file mode. I think files like the Benchmarks.manifest will not be bundled and getting the base directory and then looking for the file will work. But I'm not completely sure about UnitTest.dll if it gets bundled along with the other dlls you might not find it.

Reply:
Does `UnitTest.dll` used just because it happens to exists in the build of the test project? Maybe test can have some non-executable file which will be read. Or this code tests that executable file can be read?